### PR TITLE
Project Web Mercator Tiles to Sphere with ShaderMaterial

### PR DESCRIPTION
**Pull request by cmurphy23**
This PR is for #74, where I also wrote some comments about my work as I went.

The changes are primarily to MapSphereNode, specifically swapping the material for a ShaderMaterial that calculates the color of a given position by looking up the corresponding pixel on the web mercator tile image. In addition, we change the size of the tiles so that their position properly matches the coordinates they are supposed to show. This method, while not identical, was inspired by a very similar approach used by CesiumJS.

Added the `applyTexture` function for MapNode base class so that MapSphereNode could override it load the texture into the ShaderMaterial.

Also added some utility functions to UnitsUnitls.

I wasn't sure whether to update the build/ folder or not, I can remove those changes from the PR if necessary.

Before:
image

After: 
image

### CODE REVIEW

File: source/utils/UnitsUtils.ts [hunk 1]
```diff
@@ -140,4 +145,31 @@ export class UnitsUtils
...
 	}
+
+	public static getTileSize(zoom: number): number
+	{
+		const maxExtent = UnitsUtils.MERCATOR_MAX_EXTENT;
+		const numTiles = Math.pow(2, zoom);
+		return 2 * maxExtent / numTiles;	
+	}
+
+	public static tileBounds(zoom: number, x: number, y: number): number[]
```
Comment (tentone) [inline source/utils/UnitsUtils.ts:156]: Please add docs for these parameters.

**Review by tentone — APPROVED** [review]
Hello

Check documentation for auxiliary methods please, other than that code seems alright.

**Comment by cmurphy23** [comment]
Okay, I added some documentation for those methods, I reverted the build folder to before my changes, and I fixed some weird indentation on the shader code.

**Comment by FengFengmomo** [comment]
i have modify code as yours but there still some misalignment 
image

**Comment by cmurphy23** [comment]
I'm not sure if this PR is the right place to discuss, but the problem you're having is because the tile bounds expected by the shader code don't match the tile bounds used for the tile's geometry, specifically in the Y/latitutde direction. I recommend using identical calculations for both.

**Pull request merged**